src/media/js/elements/select.js

```diff
@@ -0,0 +1,218 @@
+                        }
+                    }
+                    if (bool) {
+                        this.classList.toggle(VISIBLE, bool);
+                    } else {
+                        this.classList.toggle(VISIBLE);
+                    }
+                    return this;
+                },
+            },
+            select: {
+                // Value is set in the createdCallback.
+                value: null,
+                writable: true
+            }
+        }),
+    });
+
+    function alignMktOptions(mktSelect) {
+        var offset = $('mkt-selected-text', mktSelect).position().left;
+        $('mkt-option', mktSelect).css('padding-left', offset);
+        return offset;
+    }
+
+    function updateMktOptionIndices(mktSelect, selectedOption) {
+        // Sets a data-index on each mkt-option that is not [selected].
+        // Prominently used to set colors since we don't have CSS4 nth-match.
+        var mktOptions = mktSelect.querySelectorAll('mkt-option');
+        var index = 0;
+        for (var i = 0; i < mktOptions.length; i++) {
+            var mktOption = mktOptions[i];
+            if (mktOption === selectedOption) {
+                mktOption.removeAttribute('data-mkt-option--index');
+            } else {
+                mktOption.setAttribute('data-mkt-option--index', index++);
+            }
+        }
+    }
+
+    function copyAttrs(dest, source) {
+        // Copies attributes from one element to another.
+        for (var i = 0; i < source.attributes.length; i++) {
+            var attr = source.attributes[i];
+            if (attr !== 'id') {
+                dest.setAttribute(attr.name, attr.value);
+            }
+        }
+        return dest;
+    }
+
+    function proxyElement(htmlElement, customElement, key) {
+        // Proxy custom element interface to actual element.
+        Object.getOwnPropertyNames(htmlElement.prototype).forEach(function(method) {
+            try {
+                customElement.prototype[method] = {
+                    value: function() {
+                        return this[key][method]();
+                    }
+                };
+            } catch (e) {
+                Object.defineProperty(customElement.prototype, method, {
+                    get: function() {
+                        return this[key][method];
+                    }
+                });
+            }
+        });
+    }
+
+    proxyElement(HTMLSelectElement, MktSelectElement, 'select');
+    proxyElement(HTMLOptGroupElement, MktOptGroupElement, 'optGroup');
+    proxyElement(HTMLOptionElement, MktOptionElement, 'option');
+
+    z.body.on('click', 'mkt-selected', function() {
+        this.parentNode.toggle();
+    })
+
+    .on('change', 'mkt-select', function() {
+        // Update <mkt-selected-text>.
+        var newText = this.querySelector('mkt-option[selected]').innerHTML;
+        this.parentNode.querySelector('mkt-selected-text').innerHTML = newText;
```

Comment:
Making a generic element would be weird since this is pulling the selected options text out, so I think that's probably as generic as it can be. I think angular 2.0 has templates that make things like this easy where you do something like `<some-element [textContent=someOtherElement.textContent]>...` and then it knows to set the property. That syntax is probably wrong but it might be neat to have an element that can easily pull data from another element on some event.